packages/tests/shadydom/event-path.html

```diff
@@ -999,17 +1125,25 @@
 
         ShadyDOM.wrap(target).dispatchEvent(new Event('some-event', {bubbles: true, composed: true}));
         assert.deepEqual(eventLog, [
-          {currentTarget: outerNoSlot, eventPhase: Event.AT_TARGET},
+          {currentTarget: fixtureRoot, eventPhase: Event.AT_TARGET},
+          {currentTarget: ShadyDOM.wrap(fixtureRoot).shadowRoot, eventPhase: Event.CAPTURING_PHASE},
           {currentTarget: outerDiv, eventPhase: Event.CAPTURING_PHASE},
           {currentTarget: innerNoSlot, eventPhase: Event.CAPTURING_PHASE},
           {currentTarget: target, eventPhase: Event.AT_TARGET},
           {currentTarget: target, eventPhase: Event.AT_TARGET},
           {currentTarget: innerNoSlot, eventPhase: Event.BUBBLING_PHASE},
           {currentTarget: outerDiv, eventPhase: Event.BUBBLING_PHASE},
-          {currentTarget: outerNoSlot, eventPhase: Event.AT_TARGET},
+          {currentTarget: fixtureRoot, eventPhase: Event.AT_TARGET},
+          // The bubbling event listener on this shadow root is incorrectly run
```

Comment:
It's unclear why the order of running the event listeners should be wrong. Can you explain that?

Reply:
When you add an event listener to a shadow root, the wrapper actually adds a (wrapped) listener to the shadow root's host:

https://github.com/webcomponents/polyfills/blob/9bd1fc0e13ba3c68d691ff7115397b6e735a8edc/packages/shadydom/src/patches/ShadowRoot.js#L21-L30

This way, even though Shady DOM's shadow roots aren't nodes in the real tree, their listeners will be still called by natively dispatched events. However, the order in which they're added ends up affecting when they're called because all the listeners added by the user to the host or its shadow root are all added to that host's single native list of listeners. For example, if bubbling listener A is first added to some host and then bubbling listener B is added to its shadow root, a bubbling, composed event dispatched to a node inside the shadow root will call A and then B because A was added first, even though B would normally be called first in a browser with native Shadow DOM.

Reply:
Ah, got it, that is not ideal but makes sense.